testutils/check-calling-form-in-baseline/main.go

```diff
@@ -175,45 +176,78 @@ type checkConfig struct {
 	form string
 }
 
-var (
-	callingFormRe = regexp.MustCompile(`calling form: "(.*?)"`)
-	valueSetRe    = regexp.MustCompile(`valueSet "(.*?)"`)
-)
-
 // deleteFoundForms reads the baseline, then deletes any used calling forms and value sets from forms.
-func deleteFoundForms(baseline string, lang string, forms map[checkConfig]bool) error {
-	// Split the baseline file into its "logical files"
-	var files []string
-	for {
-		const fileHeader = "============== file:"
-		p := strings.Index(baseline, fileHeader)
-		if p < 0 {
-			files = append(files, baseline)
-			break
-		}
-		files = append(files, baseline[:p])
+func deleteFoundForms(r io.Reader, lang string, forms map[checkConfig]bool) error {
+	var callingForm, valueSet string
 
-		baseline = baseline[p:]
-		baseline = baseline[strings.IndexByte(baseline, '\n')+1:]
+	// del deletes the form specified in callingForm, valueSet, and lang from forms and reset.
+	// We call this at the end of every logical file.
+	del := func() {
+		delete(forms, checkConfig{
+			id:   valueSet,
+			form: callingForm,
+			lang: lang,
+		})
+		callingForm = ""
+		valueSet = ""
 	}
 
-	for _, file := range files {
-		form := callingFormRe.FindStringSubmatch(file)
-		if len(form) == 0 {
+	br := bufio.NewReader(r)
+	for {
+		line, err := br.ReadString('\n')
+		if err == io.EOF {
+			del()
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+
+		if strings.HasPrefix(line, "============== file:") {
+			del()
```

Comment:
It seems to me that it's clearer to call `del()` at the bottom of the `for` loop, rather than setting up all the info needed for the delete but deferring it until the next iteration. This would also get you to have the `del()` in one place rather than two (and incidentally, would allow you to inline it if you choose)

Reply:
Oh, I see what you mean. That makes sense. It still seems like a bit of unfortunate cognitive load to handle the `del` in multiple places and at the start of the next iteration. I was hoping to differentiate reading the line from processing the file, but that seems to make the code more verbose.....

Maybe add a comment like 
```
// Handle the end of the file currently being processed
```